### PR TITLE
New folder structure & layer editor folder widget

### PR DESCRIPTION
**Pull request by FilipLeitner**
## Description

Please give a short description of the changes you propose. Please use prose and try not to be too technical, if possible. Make sure to mention people that you think should know about the PR.

## Related issues or pull requests

Please list issues or pull requests that the changes you propose are related to. It does not matter if they are still open and/or unmerged, any link is appreciated.

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Dependency updates
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe)
- [ ] I am unsure (we'll look into it together)

## Do you introduce a breaking change?

- [ ] Yes
- [ ] No
- [ ] I am unsure (no worries, we'll find out)

## Checklist

- [x] I understand and agree that the changes in this PR will be licensed under the [MIT License]
- [ ] I have followed the [guidelines for contributing](https://github.com/hslayers/hslayers-ng/blob/master/CONTRIBUTING.md)
- [ ] The proposed change fits to the content of the [code of conduct](https://github.com/hslayers/hslayers-ng/blob/master/CODE_OF_CONDUCT.md)
- [ ] I have added or updated tests and documentation, and the test suite passes (run `npm test` locally)
- [ ] I'm lost; why do I have to check so many boxes? Please help!

### CODE REVIEW

File: projects/hslayers/components/layer-manager/layer-manager.component.ts [hunk 1]
```diff
@@ -187,6 +215,104 @@ export class HsLayerManagerComponent
...
+    this.hsLayerManagerService.data.filter = this.isVisible$.pipe(
+      /**
+       * Component is constructed despite not being visible (mainPanel) on app init
+       * Thus we cannot keep constructing this stream until layermanager is actually visible
+       * (input is available)
+       */
+      filter((visible) => !!visible),
+      //Make sure filterInput is created
+      debounceTime(0),
+      //Take only the first successfull emission
```
Comment (jmacura) [inline projects/hslayers/components/layer-manager/layer-manager.component.ts:238]:
```suggestion
      //Take only the first successful emission
```

File: projects/hslayers/components/layer-manager/widgets/layer-folder-widget/layer-folder-widget.component.ts [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+import {Observable, map} from 'rxjs';
+import {getPath, setPath} from 'hslayers-ng/common/extensions';
+
+@Component({
+  selector: 'hs-layer-folder-widget',
+  standalone: true,
+  imports: [NgIf, AsyncPipe, TranslateCustomPipe, ReactiveFormsModule],
+  templateUrl: './layer-folder-widget.component.html',
+})
+export class LayerFolderWidgetComponent
```
Comment (jmacura) [inline projects/hslayers/components/layer-manager/widgets/layer-folder-widget/layer-folder-widget.component.ts:31]:
```suggestion
export class HsLayerFolderWidgetComponent
```

**Review by jmacura — COMMENTED** [review]
Would it be hard to add a drop-down list here?
image
Like an alternative -- either write or select existing.

**Comment by FilipLeitner** [comment]
> Would it be hard to add a drop-down list here? image Like an alternative -- either write or select existing.

It already kinda works that way . As it say ‘Select an option’. You can pick from existing folders or select <New folder> option which basically takes you back and changes placeholder to ‘Type new folder name’ . 
But dropdown might be still be a better option as datalist dropdown UI is not consistent across the browsers.

**Comment by jmacura** [comment]
Aye. In Firefox, at a first glance (at element focus), it looks just like a textbox.
image
Only on second click of mouse it displays a simple list, which actually looks the same as a browser form memory list.
image

Another nasty side-effect is when I choose `<new folder>` and start typing, the placeholder stays in the text-field, overlapping the typed text. Not sure if this is also a Firefox specific issue.
image

But FWIW, this PR is good to go from my side and I will approve. It switching of folders works nice, thanks! If you feel like adding another commit which will address the above issues, go ahead. Otherwise you can merge.

**Comment by FilipLeitner** [comment]
> Aye. In Firefox, at a first glance (at element focus), it looks just like a textbox. image Only on second click of mouse it displays a simple list, which actually looks the same as a browser form memory list. image
> 
> Another nasty side-effect is when I choose `<new folder>` and start typing, the placeholder stays in the text-field, overlapping the typed text. Not sure if this is also a Firefox specific issue. image
> 
> But FWIW, this PR is good to go from my side and I will approve. It switching of folders works nice, thanks! If you feel like adding another commit which will address the above issues, go ahead. Otherwise you can merge.


WOW. Hmm that’s not great. Worked much better in chrome. I’ll replace that datalist with some kind combination of dropdown  and input as suggested in separate PR/issue for 14.1

**Review by jmacura — APPROVED** [review]

**Comment by jmacura** [comment]
refs #4861

**Review by jmacura — APPROVED** [review]
The new UI works great, thanks!
image

**Pull request merged**